Building native components with `-msbuild` fails when VS 2022 is installed

I have latest dogfood version of VS 2022 installed (I also have VS 2019 SxS installed). Trying to run for example `build.cmd -vs CoreCLR.sln` fails. Trying to build any native component with `-msbuild` fails this way:

```console
CMake Error: Could not create named generator Visual Studio

Generators
* Visual Studio 17 2022        = Generates Visual Studio 2022 project files.
                                 Use -A option to specify architecture.
```

I have latest (3.21.1) CMake installed.

--- a/eng/native/gen-buildsys.cmd
+++ b/eng/native/gen-buildsys.cmd
@@ -28,6 +28,7 @@ if /i "%__Ninja%" == "1" (
     set __CmakeGenerator=Ninja
 ) else (
     if /i NOT "%__Arch%" == "wasm" (
+        if /i "%__VSVersion%" == "vs2022" (set __CmakeGenerator=%__CmakeGenerator% 17 2022)
         if /i "%__VSVersion%" == "vs2019" (set __CmakeGenerator=%__CmakeGenerator% 16 2019)
         if /i "%__VSVersion%" == "vs2017" (set __CmakeGenerator=%__CmakeGenerator% 15 2017)
 

